File: Hydroq/data/scripts/MenuBehavior.lua
 MenuBehavior = { 
   OnInit = function() 
     MenuBehavior_beh:SubscribeForMessages(StrId("BUTTON_CLICKED")) 
   end, 
   OnMessage = function(msg) 
     if(msg:HasAction(StrId("BUTTON_CLICKED")))
	 then
	   if(msg.contextNode.tag == "sgame_but")
	   then
	     CogSwitchToScene("main_menu_map", "none")
	   end
	   
	   if(msg.contextNode.tag == "multiplayer_but")
	   then
	     CogSwitchToScene("multiplayer_map", "none")
	   end
	 end
   end, 
   Update = function(delta, absolute) 
   end 
} 

MenuBehavior_beh = Behavior() 
MenuBehavior_beh:Register(MenuBehavior, "MenuBehavior")

File: Hydroq/src/MenuBehavior.cpp
#include "MenuBehavior.h"
#include "Node.h"
#include "ComponentStorage.h"
#include "Tween.h"
#include "Stage.h"

void MenuBehavior::OnInit() {
	SubscribeForMessages(ACT_BUTTON_CLICKED);
}

void MenuBehavior::OnMessage(Msg& msg) {

	if (msg.HasAction(ACT_BUTTON_CLICKED)) {
		if (msg.GetContextNode()->GetTag().compare("sgame_but") == 0) {
			// click on single game button
			auto sceneContext = GETCOMPONENT(Stage);
			// switch scene to main_menu_map with map and faction selection
			auto scene = sceneContext->FindSceneByName("main_menu_map");
			sceneContext->SwitchToScene(scene, TweenDirection::NONE);
		}
		else if (msg.GetContextNode()->GetTag().compare("multiplayer_but") == 0) {
			// click on multiplayer button
			auto sceneContext = GETCOMPONENT(Stage);
			// switch scene to multiplayer_map 
			auto scene = sceneContext->FindSceneByName("multiplayer_map");
			sceneContext->SwitchToScene(scene, TweenDirection::LEFT);
		}
	}
}